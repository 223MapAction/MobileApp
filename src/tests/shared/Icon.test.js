import React from "react";
import { render } from "@testing-library/react-native";
import IconComponent from "../../shared/Icon"; 

describe("IconComponent", () => {
    it("affiche l'icône avec les propriétés correctes lorsqu'elle est concentrée", () => {
      const { toJSON } = render(
        <IconComponent focused={true} name="home" color="#000" />
      );
  
      const tree = toJSON();
      
      // Vérifiez que l'icône est rendue avec les bonnes propriétés
      expect(tree).toMatchSnapshot();
      expect(tree.props.name).toBe("home");
      expect(tree.props.style).toMatchObject({ color: "#000" });
    });
  
    it("affiche l'icône avec les propriétés correctes lorsqu'elle n'est pas concentrée", () => {
      const { toJSON } = render(
        <IconComponent focused={false} name="user" color="#fff" />
      );
  
      const tree = toJSON();
  
      expect(tree).toMatchSnapshot();
      expect(tree.props.name).toBe("user");
      expect(tree.props.style).toMatchObject({ color: "#fff" });
    });
  });