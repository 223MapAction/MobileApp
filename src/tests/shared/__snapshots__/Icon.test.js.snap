// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent affiche l'icône avec les propriétés correctes lorsqu'elle est concentrée 1`] = `<Text />`;

exports[`IconComponent affiche l'icône avec les propriétés correctes lorsqu'elle n'est pas concentrée 1`] = `
<Text
  allowFontScaling={false}
  focused={false}
  selectable={false}
  style={
    [
      {
        "color": undefined,
        "fontSize": 24,
      },
      {
        "color": "#fff",
      },
      {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
  testID="icon"
>
  
</Text>
`;
